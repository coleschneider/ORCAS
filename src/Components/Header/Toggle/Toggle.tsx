import * as React from 'react';

const Toggle = ({ isOpen, onClick }) => (
  <div
    data-testid="toggle"
    className={`header-menu-toggle ${isOpen && 'is-clicked'}`}
    id="header-menu-toggle-it"
    onClick={onClick}
  >
    <span>Menu</span>
  </div>
);
export default Toggle;
