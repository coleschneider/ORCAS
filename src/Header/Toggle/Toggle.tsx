import * as React from 'react';

const Toggle = ({ isOpen, isMobile, onClick }) => (
  <div
    data-testid="toggle"
    className={`header-menu-toggle ${isOpen && 'is-clicked'}`}
    id="header-menu-toggle-it"
    style={{ display: isMobile ? 'block' : undefined }}
    onClick={onClick}
  >
    <span>Menu</span>
  </div>
);
export default Toggle;
