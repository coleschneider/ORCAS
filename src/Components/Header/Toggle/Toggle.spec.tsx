import * as React from 'react';
import Toggle from './Toggle';

describe('Toggle', () => {
  const props = {
    isMobile: false,
    isOpen: false,
    onClick: jest.fn(),
  };
  let wrapper;
  const setup = (override = {}) => (wrapper = shallow(<Toggle {...props} {...override} />));

  it('matches the snapshot', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('displays properly when isOpen is false', () => {
    setup();
    expect(wrapper.hasClass('is-clicked')).toBe(false);
  });
  it('displays properly when isOpen is true', () => {
    setup({ isOpen: true });
    expect(wrapper.hasClass('is-clicked')).toBe(true);
  });
  it('calls onClick correctly', () => {
    const mockClick = jest.fn();
    setup({ onClick: mockClick });
    expect(mockClick.mock.invocationCallOrder).toMatchObject([]);
    wrapper.find('[test-id="toggle"]').simulate('click');
    expect(mockClick.mock.invocationCallOrder[0]).toBe(1);
  });
});
