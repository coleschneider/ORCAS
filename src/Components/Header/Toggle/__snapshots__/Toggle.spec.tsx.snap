// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle matches the snapshot 1`] = `
<div
  className="header-menu-toggle false"
  id="header-menu-toggle-it"
  onClick={[MockFunction]}
  test-id="toggle"
>
  <span>
    Menu
  </span>
</div>
`;
